cub/cub/util_type.cuh

```diff
@@ -209,14 +209,28 @@ struct NullType
  * dispatch based on constant integral values)
  */
 template <int A>
-struct Int2Type
+struct CCCL_DEPRECATED_BECAUSE("Use ::cuda::std::integral_constant instead") Int2Type
 {
   enum
   {
     VALUE = A
   };
 };
 
+namespace detail
+{
+
+template <bool Value>
+inline constexpr auto bool_constant_v = ::cuda::std::bool_constant<Value>{};
```

Comment:
Since we have `constant_v`, we shouldn't need `bool_constant_v` anymore. Please remove it for consistency.
```suggestion
template <bool Value>
inline constexpr auto bool_constant_v = ::cuda::std::bool_constant<Value>{};
```

Reply:
I think I understand why you have it, so you can save a cast in case of `constant_v<(bool)integral_value>`. Alright then.